README.md

```diff
@@ -34,12 +34,27 @@ Learn more at [ipfs.io](https://ipfs.io) (it is really cool, we promise!)
 
 #### Automagical Detection of IPFS Resources
 
-  Requests for resources at IPFS-like paths (`/ipfs/$cid` or `/ipns/$peerid_or_fqdn`) are detected on any website.  
+  Requests for IPFS-like paths (`/ipfs/$cid` or `/ipns/$peerid_or_fqdn-with-dnslink`) are detected on any website.  
   If tested path is a valid IPFS address it gets redirected and loaded from a local gateway, e.g:  
   `https://ipfs.io/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`  
   → `http://127.0.0.1:8080/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`
 
-#### IPFS Status and Action Menu
+#### IPFS API exposed as `window.ipfs`
+
+A safe subset of IPFS APIs of user's node is exposed as `window.ipfs` on every webpage.  
+This means websites can detect if `window.ipfs` exists and opt-in to use it instead of spawning own `js-ipfs` node.  This saves resources, battery etc.
+Make sure to read our [notes on `window.ipfs`](https://github.com/ipfs-shipyard/ipfs-companion/blob/master/docs/window.ipfs.md), where we explain it in-depth and provide examples on how to use it your own dapp.
+
+#### Embedded `js-ipfs` Node
+
+Browser Action menu provides a toggle for switching  between embedded, in-memory `js-ipfs` and external IPFS node accessed over HTTP API.
+
+Embedded node is a fine fallback for situations when user needs to quickly share a file with someone or just wants to run a dapp against `window.ipfs` but has no third party IPFS daemon running locally.
+Power users can provide own config (eg. to enable experimental pubsub) via _Preferences_.
+
+**UX Note:** Embedded node does not run when external one is used. Every time you switch back to Embedded node, a new instance is spawned on-demand, so it takes [a few seconds](https://user-images.githubusercontent.com/157609/38493690-4a77bd9e-3bf3-11e8-85da-ba06fd94cdbf.gif) for a brand-new node to find peers.
```

Comment:
I don't think we need to specify that this is a "UX" note...I think "Note" would do.

"**Note:** The embedded node _does not run_ when external node is selected. Every time you switch back to the embedded node, a new instance is created on-demand. It can take [a few seconds](https://user-images.githubusercontent.com/157609/38493690-4a77bd9e-3bf3-11e8-85da-ba06fd94cdbf.gif) for a brand-new node to find peers."